### PR TITLE
[diffusion] Fuse Wan VAE causal conv padding

### PR DESCRIPTION
**Pull request by mickqian**
## Summary

Fuse the Wan VAE causal-conv cache concat and spatial padding path with a Triton kernel.

The fast path is conservative: CUDA-only, contiguous NCDHW tensors, cached causal conv, symmetric spatial padding, and non-empty spatial padding. Other cases keep the existing `torch.cat` + `F.pad` fallback.

## Experiments

- H200 source-level exact check over representative Wan causal conv shapes:
  - fp32 and bf16
  - padded tensor max diff: `0`
  - conv output max diff: `0`
- 1GPU LingBot realtime consistency:
  - `lingbot_world_realtime_plastic_beach`: passed
  - CLIP similarity: `1.0000`
  - SSIM: `1.0000`
  - PSNR: `inf`
  - max mean abs diff: `0.0000`
- H200 bf16 microbench on cached 3x3x3 Wan causal conv shapes:
  - fused cat+pad speedup: `1.49x` to `2.30x`
  - cat+pad+conv speedup: `1.14x` to `1.31x`
  - max diff: `0`
  - 3x1x1 and no-cache cases are intentionally not fused because they showed no reliable win
- 4xH200 LingBot realtime, same environment, averaged over stable chunks 2 and 3 from two baseline runs and two patched runs:

| metric | baseline | patched | delta |
| --- | ---: | ---: | ---: |
| VAE decode | 222.2 ms | 205.7 ms | -16.5 ms / -7.4% |
| scheduler forward | 1410.0 ms | 1394.5 ms | -15.5 ms / -1.1% |
| chunk total | 1485.0 ms | 1467.0 ms | -18.0 ms / -1.2% |

---
### CI States

Latest PR Test (Base): :x: [Run #27365188403](https://github.com/sgl-project/sglang/actions/runs/27365188403)
Latest PR Test (Extra): :x: [Run #27365187441](https://github.com/sgl-project/sglang/actions/runs/27365187441)

### CODE REVIEW

File: python/sglang/jit_kernel/diffusion/triton/wan_vae_pad.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    x_t = src_t - cache_t
+    x_offsets = (((ob * channels + oc) * t_size + x_t) * h_size + ih) * w_size + iw
+    cache_offsets = (
+        ((ob * channels + oc) * cache_t + src_t) * h_size + ih
+    ) * w_size + iw
+
+    x_vals = tl.load(x_ptr + x_offsets, mask=valid & ~from_cache, other=0.0)
+    cache_vals = tl.load(cache_ptr + cache_offsets, mask=valid & from_cache, other=0.0)
+    vals = tl.where(from_cache, cache_vals, x_vals)
+    tl.store(out_ptr + offsets, vals, mask=mask)
```
[the remark this answers is not in the document]
Comment (mickqian) [reply python/sglang/jit_kernel/diffusion/triton/wan_vae_pad.py:63]: Keeping this kernel shape-specialized intentionally. The Wan VAE serving path uses a fixed small set of layer shapes, and making the shape/divisor values runtime scalars would add overhead in the steady-state kernel. The cache/x loads are already guarded by mutually exclusive masks, so the invalid branch offsets are not loaded; adding clamps would add extra per-element work on the hot path.

File: python/sglang/multimodal_gen/runtime/models/vaes/parallel/wan_common_utils.py [hunk 2]
```diff
@@ -6,6 +6,9 @@
 
 from sglang.multimodal_gen.runtime.platforms import current_platform
 
+if current_platform.is_cuda():
+    from sglang.jit_kernel.diffusion.triton.wan_vae_pad import fused_causal_conv3d_cat_pad
```
[the remark this answers is not in the document]
Comment (mickqian) [reply python/sglang/multimodal_gen/runtime/models/vaes/parallel/wan_common_utils.py]: No fallback binding is needed here. The fused function is only referenced after `_can_fuse_causal_conv3d_cat_pad` has returned true, and that helper returns false before the call on non-CUDA platforms. Keeping this as a CUDA-only import avoids an unused non-CUDA symbol.

**Comment by mickqian** [comment]
duplicated with #28204
